src/manifest/dbtProject.ts

```diff
@@ -578,6 +579,56 @@ select * from renamed
     this.queryResultPanel.getSummary(compiledSql, this.getAdapterType());
   }
 
+  private getLimitQuery(queryTemplate: string, query: string, limit: number) {
+    return queryTemplate
+      .replace("{query}", () => query)
+      .replace("{limit}", () => limit.toString());
+  }
+
+  private async getQuery(
+    query: string,
+    limit: number,
+  ): Promise<{ queryTemplate: string; limitQuery: string }> {
+    const queryTemplate = workspace
+      .getConfiguration("dbt")
+      .get<string>("queryTemplate");
+
+    if (queryTemplate) {
+      console.log("Using user provided query template", queryTemplate);
+      const limitQuery = this.getLimitQuery(queryTemplate, query, limit);
+
+      return { queryTemplate, limitQuery };
+    }
+
+    try {
+      const dbtVersion = await this.getDBTVersion();
+      //dbt supports limit macro after v1.5
+      if (dbtVersion && dbtVersion[0] >= 1 && dbtVersion[1] >= 5) {
+        const args = { sql: query, limit };
+        const queryTemplateFromMacro = await this.python?.lock(
+          (python) =>
+            python!`to_dict(project.execute_macro('get_limit_subquery_sql', ${args}))`,
+        );
+
+        console.log("Using query template from macro", queryTemplateFromMacro);
+        return {
+          queryTemplate: queryTemplateFromMacro,
+          limitQuery: queryTemplateFromMacro,
+        };
+      }
+    } catch (err) {
+      this.telemetry.sendTelemetryError(
+        "executeMacroGetLimitSubquerySQLError",
+        err,
+        { adapter: this.adapterType },
+      );
+    }
+    return {
+      queryTemplate: DefaultQueryTemplate,
+      limitQuery: this.getLimitQuery(DefaultQueryTemplate, query, limit),
+    };
+  }
```

Comment:
This shold not be in dbtProject but in dbtCoreIntegration.

Reply:
fix it, pls re review